src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -1732,37 +1756,39 @@ subroutine ice_shelf_advect_thickness_x(CS, G, LB, time_step, hmask, h0, h_after
 
   do j=jsh,jeh ; do I=ish-1,ieh
     if (CS%u_face_mask(I,j) == 4.) then ! The flux itself is a specified boundary condition.
-      uh_ice(I,j) = time_step * G%dyCu(I,j) * CS%u_flux_bdry_val(I,j)
+      uh_ice(I,j) = (time_step * G%dyCu(I,j)) * CS%u_flux_bdry_val(I,j)
     elseif ((hmask(i,j) == 1 .or. hmask(i,j) == 3) .or. (hmask(i+1,j) == 1 .or. hmask(i+1,j) == 3)) then
       u_face = 0.5 * (CS%u_shelf(I,J-1) + CS%u_shelf(I,J))
       h_face = 0.0 ! This will apply when the source cell is iceless or not fully ice covered.
 
       if (u_face > 0) then
         if (hmask(i,j) == 3) then ! This is a open boundary inflow from the west
-          h_face = CS%thickness_bdry_val(i,j)
+          h_face = CS%h_bdry_val(i,j)
         elseif (hmask(i,j) == 1) then ! There can be eastward flow through this face.
-          if ((hmask(i-1,j) == 1) .and. (hmask(i+1,j) == 1)) then
+          if ((hmask(i-1,j) == 1 .or. hmask(i-1,j) == 3) .and. &
+            (hmask(i+1,j) == 1 .or. hmask(i+1,j) == 3)) then
             slope_lim = slope_limiter(h0(i,j)-h0(i-1,j), h0(i+1,j)-h0(i,j))
             ! This is a 2nd-order centered scheme with a slope limiter.  We could try PPM here.
-            h_face = h0(i,j) - slope_lim * 0.5 * (h0(i,j)-h0(i+1,j))
+            h_face = h0(i,j) - slope_lim * (0.5 * (h0(i,j)-h0(i+1,j)))
```

Comment:
Multiplying by an integer power of 2 (e.g., 0.5 here) is order invariant, so the extra parentheses here are unnecessary but benign.